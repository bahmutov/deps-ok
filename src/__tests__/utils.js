const cleanVersion = require('../utils').cleanVersion
const getAllDependencies = require('../utils').getAllDependencies

/* eslint-env jest */
describe('clean version', () => {
  it('^ symbol', () => {
    const version = cleanVersion('^0.1.1')
    expect(version).toBe('0.1.1')
  })
})

describe('getAllDependencies', () => {
  it('^ symbol', () => {
    const pkg = {
      dependencies: {
        foo: '^0.1.0'
      }
    }
    const deps = getAllDependencies(pkg)
    expect(deps).toMatchSnapshot()
  })

  it('returns all 3 types', () => {
    const pkg = {
      dependencies: {
        foo: '0.1.0'
      },
      devDependencies: {
        bar: 'latest'
      },
      peerDependencies: {
        baz: '2.0.0'
      }
    }
    const deps = getAllDependencies(pkg)
    expect(deps).toMatchSnapshot()
  })

  it('duplicate', () => {
    const pkg = {
      dependencies: {
        foo: '0.1.0'
      },
      devDependencies: {
        foo: '2.0.0'
      }
    }
    expect(() => getAllDependencies(pkg)).toThrowErrorMatchingSnapshot()
  })

  it('allowed duplicate', () => {
    const pkg = {
      dependencies: {
        foo: '0.1.0'
      },
      devDependencies: {
        foo: '2.0.0'
      }
    }
    const options = {
      allowDuplicate: ['foo']
    }
    getAllDependencies(pkg, options)
  })
})
