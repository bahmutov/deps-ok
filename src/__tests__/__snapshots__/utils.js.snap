// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getAllDependencies ^ symbol 1`] = `
Object {
  "foo": "^0.1.0",
}
`;

exports[`getAllDependencies duplicate 1`] = `"duplicate properties found: foo"`;

exports[`getAllDependencies returns all 3 types 1`] = `
Object {
  "bar": "latest",
  "baz": "2.0.0",
  "foo": "0.1.0",
}
`;
